tm2py/components/network/highway/highway_assign.py

```diff
@@ -152,17 +152,16 @@ def run(self):
                     None
                 self._create_skim_matrices(scenario, assign_classes)
 
-                if iteration == 0 or not run_dynamic_toll:
-                    assign_spec = self._get_assignment_spec(assign_classes)
+                assign_spec = self._get_assignment_spec(assign_classes)
+
+                if not run_dynamic_toll:
```

Comment:
Missing condition: If warmstart is turned off, iteration 0 does not run dynamic tolling.

Reply:
updated in 178cd28